### ISSUE TITLE
go panic: interface conversion: schema.Type is *schema.MapType, not *schema.OptionalType

### ISSUE THREAD

**Issue by donaldong** [pulumi/pulumi#7733]
pulumi version: v3.9.1

## Steps to reproduce
Import a cloudfront distribution using `pulumi import`.

See the following stack trace
```
panic: interface conversion: schema.Type is *schema.MapType, not *schema.OptionalType

goroutine 129 [running]:
github.com/pulumi/pulumi/pkg/v3/codegen/schema.(*types).bindProperties(0xc000b11030, 0xc0034af3b0, 0xc00055f300, 0xc, 0x10, 0xc003057d00, 0xc0030893c0, 0x6, 0x0, 0x0, ...)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/codegen/schema/schema.go:2005 +0xdd5
github.com/pulumi/pulumi/pkg/v3/codegen/schema.bindResource(0xc003679520, 0x19, 0xc0036e4000, 0x342f, 0xc0034af3b0, 0x0, 0x0, 0xc00055f300, 0xc, 0x10, ...)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/codegen/schema/schema.go:2252 +0x132
github.com/pulumi/pulumi/pkg/v3/codegen/schema.bindResources(0xc003057c20, 0xc000b11030, 0xc0063c97d0, 0x17c, 0xc003057bc0, 0x0, 0x0, 0x0, 0x0)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/codegen/schema/schema.go:2349 +0x1b8
github.com/pulumi/pulumi/pkg/v3/codegen/schema.importSpec(0xc0008394fc, 0x3, 0xc000839508, 0x6, 0xc00186c540, 0x55, 0xc001dadd60, 0x2, 0x2, 0xc0006d8cf0, ...)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/codegen/schema/schema.go:1418 +0x548
github.com/pulumi/pulumi/pkg/v3/codegen/schema.(*pluginLoader).LoadPackage(0xc001641b30, 0xc00004a10a, 0x3, 0x0, 0x0, 0x0, 0x0)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/codegen/schema/loader.go:97 +0x30f
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*ImportStep).Apply(0xc000692070, 0x1, 0x5a141ba, 0x23, 0xc001533e30, 0x3)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/resource/deploy/step.go:925 +0x464
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*stepExecutor).executeStep(0xc0005ed500, 0x3c, 0x5d94398, 0xc000692070, 0x0, 0x0)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/resource/deploy/step_executor.go:271 +0x209
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*stepExecutor).executeChain(0xc0005ed500, 0x3c, 0xc0014ea740, 0x1, 0x1)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/resource/deploy/step_executor.go:221 +0xef
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*stepExecutor).worker.func1(0xc0005ed500, 0x3c, 0xc0014ea740, 0x1, 0x1, 0xc0015a4900)
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/resource/deploy/step_executor.go:377 +0xc5
created by github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*stepExecutor).worker
        /private/tmp/pulumi-20210730-80952-1hmuscx/pkg/resource/deploy/step_executor.go:374 +0x2c8
```

**Comment by emiliza** [comment]
Sounds really similar to #7717 - @pgavlin can you confirm if Komal's fix (to be released tomorrow) would address this?

**Comment by emiliza** [comment]
@donaldong This should be fixed in v3.10.0+ - can you try updating and let me know if this is still reproducible for you?  Thanks!
